src/Microsoft.Identity.Client/ApiEvent.cs

```diff
@@ -0,0 +1,48 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+
+namespace Microsoft.Identity.Client
+{
+    internal class ApiEvent : Event
+    {
+        public ApiEvent(string authority, string authorityType, string uiBehavior, string validationStatus, bool loginHint = false) : base("Microsoft.MSAL.api_event")
+        {
+            this["authority"] = authority;
+            this["authority_type"] = authorityType;
+            this["ui_behavior"] = uiBehavior;
+            this["validation_status"] = validationStatus;
+            this["login_hint"] = loginHint.ToString();
+        }
+
+        public void Update(string idToken, bool wasSuccessful)
+        {
+            this["id_token"] = "TODO";  // TODO
```

Comment:
No, we are NOT sending back id_token, but we are going to send `tenant_id` and `user_id` though.

That TODO comment refers to parse out the `tenant_id` and `user_id`. This can either be implemented inside this class, or implemented by the caller of this class. We will see which one makes more sense when the usage pattern of event classes become clear.